code/game/machinery/hologram.dm

```diff
@@ -41,313 +37,274 @@ var/const/HOLOPAD_MODE = RANGE_BASED
 	idle_power_usage = 5
 	use_power = 1
 
-	var/list/mob/living/silicon/ai/masters = new() //List of AIs that use the holopad
-	var/last_request = 0 //to prevent request spam. ~Carn
-	var/holo_range = 5 // Change to change how far the AI can move away from the holopad before deactivating
-	var/hacked = 0 // if hacked the hologram doesn't display. It only gives audio feedback.
-
-	var/incoming_connection = 0
-	var/mob/living/caller_id
-	var/obj/machinery/hologram/holopad/sourcepad
-	var/obj/machinery/hologram/holopad/targetpad
-	var/forced
-	var/last_message
+	var/holopad_id = ""
 
-/obj/machinery/hologram/holopad/check_eye(mob/user)
-	if (user && user == caller_id)
-		return 0
+	var/list/active_holograms
+	var/last_request = 0 //to prevent request spam. ~Carn
+	var/holo_range = 7 // Change to change how far the AI can move away from the holopad before deactivating
 
-	return -1
+	var/incoming_connection = FALSE
+	var/established_connection = FALSE
+	var/obj/machinery/hologram/holopad/connected_pad
+	var/forced = FALSE
+	var/hacked = FALSE
+	var/last_message
 
 /obj/machinery/hologram/holopad/Initialize()
 	. = ..()
-	desc = "It's a floor-mounted device for projecting holographic images. Its ID is '[loc.loc]'"
+	var/area/A = get_area(src)
+	holopad_id = A.name
+	desc += " Its ID is '[holopad_id]'"
+
+/obj/machinery/hologram/holopad/update_icon()
+	if(length(active_holograms) || connected_pad)
+		icon_state = "holopad1"
+		set_light(2)
+	else
+		icon_state = "holopad0"
+		set_light(0)
 
 /obj/machinery/hologram/holopad/attack_hand(var/mob/living/carbon/human/user) //Carn: Hologram requests.
 	if(!istype(user))
 		return
-	if(incoming_connection && caller_id)
-		visible_message("The pad hums quietly as it establishes a connection.")
-		if(caller_id.loc != sourcepad.loc)
-			visible_message("The pad flashes an error message. The caller has left their holopad.")
-			return
-		take_call(user)
+
+	if(incoming_connection)
+		audible_message("The pad hums quietly as it establishes a connection.")
+		take_call()
 		return
-	else if(caller_id && !incoming_connection && forced)
-		audible_message("Access denied. Terminating a command-level transmission locally is not permitted.")
+	else if(connected_pad)
+		if(forced)
+			audible_message("Access denied. Terminating a command-level transmission locally is not permitted.")
```

Comment:
Should this one be a SPAN_WARNING maybe?

Reply:
nah, i think this is fine. especially if some madman intends to add Say() for machinery